Framework/PythonInterface/plugins/algorithms/PeakMatching.py

```diff
@@ -0,0 +1,307 @@
+            doc='Name of the table containing the secondary peak matches')
+
+        self.declareProperty(
+            ITableWorkspaceProperty(name='SortedByEnergy',
+                                defaultValue='all_matches_sorted_by_energy',
+                                direction=Direction.Output),
+        doc='Name of the table containing all of the peak matches sorted by energy')
+
+        self.declareProperty(
+            ITableWorkspaceProperty(name='ElementLikelyhood',
+                                    defaultValue='element_likelyhood',
+                                    direction=Direction.Output),
+            doc='Name of the table containing the count of elements in all matches')
+
+    def get_default_peak_data(self):
+        path = os.path.join(os.path.dirname(ElementalAnalysis.__file__), "peak_data.json")
+        with open(path, 'r') as f:
+            peak_data = json.load(f)
+        return peak_data
+
+    def PyExec(self):
+        table = self.getPropertyValue("PeakTable")
+        path = self.getPropertyValue("PeakDatabase")
+        peak_data = self.process_peak_data(path)
+
+        sigma_column = self.getProperty("SigmaColumn").value
+        peak_column = self.getProperty("PeakCentreColumn").value
+        input_peaks = self.get_input_peaks(table,peak_column,sigma_column)
+
+        primary_data, secondary_data, all_data = self.get_matches(peak_data, input_peaks)
+        names = [self.getPropertyValue("PrimaryPeaks"),
+                 self.getPropertyValue("SecondaryPeaks"),
+                 self.getPropertyValue("AllPeaks"),
+                 self.getPropertyValue("SortedByEnergy"),
+                 self.getPropertyValue("ElementLikelyhood")]
+        self.output_data(primary_data, secondary_data, all_data,names)
+
+    def process_peak_data(self,path):
+        '''
+        opening the json file containing the peak data
+        Format of file:
+            {
+            'Element': {
+                'Z': z,
+                'A': a,
+                'Primary' : {
+                    'Transition':energy,
+                    ..., ...
+                    },
+                'Secondary': {
+                    'Transition': energy,
+                    ..., ...
+                    },
+                'Gammas': {
+                    'Isotope': energy,
+                    ..., ...
+                    },
+                'Electron': {
+                    'Energy': probability,
+                    ..., ...
+                    }
+                '''
+        peak_data = None
+        if path:
+            with open(path, 'r') as f:
+                peak_data = json.load(f)
+        else:
+            peak_data = self.get_default_peak_data()
+
+        primary_energies = {}
+
+        secondary_energies = {}
+
+        all_energies = {}
+
+        for element in peak_data:
+            primary_energy = [float(x[1]) for x in list(peak_data[element]['Primary'].items())]
```

Comment:
can we better names then `x`? 

Also we should make the labels (e.g. `Primary`) constants at the top of the file. Then if it changes in future it is easy to update